### DISCUSSION TITLE
Details for reason why a reusable workflows was skipped?

### DISCUSSION THREAD

**Discussion by jenstroeger** [community/community#48632] [Actions] [answered]
### Select Topic Area

Question

### Body

I have a [reusable workflow that’s being skipped](https://github.com/jenstroeger/python-package-template/actions/runs/3795657881/jobs/6455035647) and I can not figure out _why_. The only feedback I get is `This job was skipped` and I can’t find logs or other information that would help me find out the reasons for skipping.

Am I missing something that would be helpful in this situation?

@github-staff would it be possible to add information as to what the “skip condition” was which evaluated true and thereby lead to skipping a workflow?

**Comment by airtower-luna** [comment, marked as answer]
The [`if` condition in the one job of the called workflow](https://github.com/jenstroeger/python-package-template/blob/0b40f48f607bf1a69b8f4963dea3105ee0f3bf93/.github/workflows/_wiki-documentation.yaml#L34) is one that'll always evaluate to false:

```yaml
    if: github.repository.has_wiki == true
```

`github.repository.has_wiki` is not a valid field, `github.repository` is simply a string with the repository name ([see `github` context docs](https://docs.github.com/en/actions/learn-github-actions/contexts#github-context)). For reasons I don't know an unset variable is treated like an empty string instead of throwing an error, so that condition turns into `'' == true`, which is obviously `false`.

Debug logging might help with future issues, see: https://docs.github.com/en/actions/monitoring-and-troubleshooting-workflows/enabling-debug-logging

**Reply by jenstroeger** [reply]
Interestingly, [`actionlint`](https://github.com/rhysd/actionlint) also caught this problem:
```
.github/workflows/_wiki-documentation.yaml:34:9: receiver of object dereference "has_wiki" must be type of object but got "string" [expression]
   |
34 |     if: github.repository.has_wiki == true
   |         ^~~~~~~~~~~~~~~~~~~~~~~~~~

```

**Comment by jenstroeger** [comment]
@airtower-luna hmm, thank you!

So, then it looks like the `repository` object changes shape depending on context? For example, when triggered by a [`release`](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#release) then
```yaml
if: github.event.repository.has_wiki == true
```
actually works.

**Comment by airtower-luna** [comment]
`github.repository` and `github.event.repository` are very different things. `github.event` is an object which can contain more objects (depending on the event type). If the event payload is what you're after the core of the problem is that in that workflow the `event` part is missing. :wink:

**Comment by jenstroeger** [comment]
I think I extrapolated that `github.event.repository`’s shape is equal to `github.repository`’s (probably by rendering the same object in the same way). Too many assumptions… 🤓

Let me try your suggestion above, and remove the `if:`.
